samcli/lib/build/app_builder.py

```diff
@@ -208,39 +217,52 @@ def update_template(template_dict: Dict, original_template_path: str, built_arti
 
         for logical_id, resource in template_dict.get("Resources", {}).items():
 
-            if logical_id not in built_artifacts:
-                # this resource was not built. So skip it
+            full_path = get_full_path(stack_path, logical_id)
+            is_artifact = full_path in built_artifacts
+            is_stack = full_path in stack_output_template_path_by_stack_path
+
+            if not is_artifact and not is_stack:
+                # this resource was not built or a nested stack. So skip it
```

Comment:
TBH, I would prefer if we choose a better name of `is_artifact` rather than this overly obvious comment. 😋

Or at least provide a reason of why `skip it`?

Reply:
comment updated!